Document/0x05j-Testing-Resiliency-Against-Reverse-Engineering.md

```diff
@@ -40,6 +43,32 @@ The following is a sample attestation result:
 }
 ```
 
+**ctsProfileMatch Vs basicIntegrity**
+
+The SafetyNet Attestation API initially provided a single value called basicIntegrity to help developers determine the integrity of a device. As the API evolved, Google introduced a new, stricter check whose results appear in a value called ctsProfileMatch, which allows developers to more finely evaluate the devices on which their app is running.
+
+In broad terms, basicIntegrity gives you a signal about the general integrity of the device and its API. Rooted devices fail basicIntegrity, as do emulators, virtual devices, and devices with signs of tampering, such as API hooks.
+
+On the other hand, ctsProfileMatch gives you a much stricter signal about the compatibility of the device. Only unmodified devices that have been certified by Google can pass ctsProfileMatch. Devices that will fail ctsProfileMatch include the following:
+
+- Devices that fail basicIntegrity
+- Devices with an unlocked bootloader
+- Devices with a custom system image (custom ROM)
+- Devices for which the manufactured didn't apply for, or pass, Google certification
+- Devices with a system image built directly from the Android Open Source Program source files
+- Devices with a system image distributed as part of a beta or developer preview program (including the Android Beta Program)
+
+**Do's and Don'ts of** `SafetyNetApi.attest`
+
+-  Create a large (16 bytes or longer) random number on your server using a cryptographically-secure random function so that a malicious user can not reuse a successful attestation result in place of an unsuccessful result
+- Trust APK information (apkPackageName, apkCertificateDigestSha256 & apkDigestSha256) only if the value of ctsProfileMatch is true.
+- Using a secure connection, entire JWS response should be send to your server, for verification. It isn't recommend to perform the verification directly in the app because, in that case, there is no guarantee that the verification logic itself hasn't been modified.
+- The verify() method only validates that the JWS message was signed by SafetyNet. It doesn't verify that the payload of the verdict matches your expectations. Also, it has a fixed rate limit of 10,000 requests per day, per project. Hence, use the verify() method only for testing during the initial development stage and not in a production scenario.

```

Comment:
is this really a good recommendation? what if we would for instance randomly select devices to do attestation? And note that if you have a high risk app, it might be worth filling in the quota form to actually raise the bar if necessary.

Reply:
I agree. How useful is attestation if you only do it in dev?

Reply:
@commjoen @TheDauntless Through the aforementioned statements, I was referring to the limited quota availibility of the `verify` method. Obviously, the signature should be verified to detect tampering using the standard cryptographic techniques [(SafetyNet Verification Samples)](https://github.com/googlesamples/android-play-safetynet/tree/master/server/java/src/main/java). 
The method with limited quota is meant for early testing phases wherein everything is not developed and since the quota is limited it shouldn't be used for the prod environment. 

I'll update the content in PR so that there's no ambiguity. Let me know if you need any further information.